File: Agile Development Practices/Code/DotNet/videoworld/UnitTests/Models/RegularPriceTest.cs
using NUnit.Framework;
using VideoWorld.Models;

namespace UnitTests.Models
{
    class RegularPriceTest
    {
        private static readonly IPrice RegularPriceInstance = new RegularPrice();

        [Test]
        public void ShouldCalculateCorrectChargeForRegularMovie()
        {
            Assert.AreEqual(2.00m, RegularPriceInstance.GetCharge(1));
            Assert.AreEqual(2.00m, RegularPriceInstance.GetCharge(2));
            Assert.AreEqual(3.50m, RegularPriceInstance.GetCharge(3));
            Assert.AreEqual(5.00m, RegularPriceInstance.GetCharge(4));
        }
    }
}

File: Agile Development Practices/Code/Java/videoworld/src/unit/java/com/thoughtworks/videorental/domain/RegularPriceTest.java
package com.thoughtworks.videorental.domain;

import static junit.framework.Assert.*;

import org.junit.Test;

public class RegularPriceTest {
	private static final RegularPrice REGULAR_PRICE = new RegularPrice();

	@Test
    public void shouldCalculateCorrectChargeForRegularMovie() throws Exception {
    	assertEquals(2.0, REGULAR_PRICE.getCharge(1));
    	assertEquals(2.0, REGULAR_PRICE.getCharge(2));
    	assertEquals(3.5, REGULAR_PRICE.getCharge(3));
    	assertEquals(5.0, REGULAR_PRICE.getCharge(4));
    }
}
